pkg/controller/handler.go

```diff
@@ -128,12 +128,51 @@ func (c *Controller) handleVirtualService(ingress *networkingv1.Ingress) error {
 		uvs.ObjectMeta.Annotations = nvs.ObjectMeta.Annotations
 		uvs.Spec = nvs.Spec
 
-		_, err = c.istioclientset.NetworkingV1beta1().VirtualServices(ingress.Namespace).Update(ctx, uvs, metav1.UpdateOptions{})
+		vs, err = c.istioclientset.NetworkingV1beta1().VirtualServices(ingress.Namespace).Update(ctx, uvs, metav1.UpdateOptions{})
 		if err != nil {
 			return err
 		}
 	}
 
+	// Check if we have an IP address on the first gateway associated
+	// with the  Virtual Service and sync it to the Ingress.
+	oid := strings.Split(vs.Spec.Gateways[0], "/")
+	if len(oid) == 1 {
+		oid[1] = oid[0]
+		oid[0] = vs.Namespace
+	}
```

Comment:
Oh I get it... The comment is for the entire next block? Maybe it would make sense to put this into its own function to seperate the steps in the process?

Reply:
I'm still trying to figure out the meaning of `oid`...

Reply:
Bahaha! Gotcha! That makes sense.